src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -992,5 +1009,21 @@ internal abstract partial class AbstractReferenceFinder<TSymbol> : AbstractRefer
             var tokens = root.DescendantTokens(descendIntoTrivia: true);
             return await FindReferencesInTokensAsync(document, semanticModel, tokens, tokensMatch, symbolsMatchAsync, cancellationToken).ConfigureAwait(false);
         }
+
+        protected static async Task<ImmutableArray<string>> GetAllMatchingGlobalAliasNamesAsync(
+            Project project, string name, int arity, CancellationToken cancellationToken)
+        {
+            using var result = TemporaryArray<string>.Empty;
+
+            var syntaxFacts = project.GetRequiredLanguageService<ISyntaxFactsService>();
+            foreach (var document in await project.GetAllRegularAndSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false))
+            {
+                var index = await SyntaxTreeIndex.GetRequiredIndexAsync(document, cancellationToken).ConfigureAwait(false);
+                foreach (var alias in index.GetGlobalAliases(name, arity))
+                    result.Add(alias);
+            }
+
+            return result.ToImmutableAndClear();
+        }
```

Comment:
this is the helper that actually tells you what global aliases there potentially are for a given symbol